src/modules/commander/Commander.cpp

```diff
@@ -1620,8 +1641,12 @@ Commander::run()
 			if (is_vtol(&status)) {
 
 				// Check if there has been any change while updating the flags
-				if (status.is_rotary_wing != vtol_status.vtol_in_rw_mode) {
-					status.is_rotary_wing = vtol_status.vtol_in_rw_mode;
+				bool is_rotary = status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING;
+
+				if (is_rotary != vtol_status.vtol_in_rw_mode) {
```

Comment:
This check is to see if the status has changed. So I think you want something like this:

```
new_vehicle_type = vtol_status.vtol_in_rw_mode ? vehicle_status_s::VEHICLE_TYPE_ROTARY_WING : vehicle_status_s::VEHICLE_TYPE_FIXED_WING;
if (new_vehicle_type != status.vehicle_type) {
    status.vehicle_type = new_vehicle_type;
```